internal/pkg/cli/init.go

```diff
@@ -115,16 +119,25 @@ func newInitOpts(vars initVars) (*initOpts, error) {
 		initEnvVars: initEnvVars{
 			GlobalOpts:   NewGlobalOpts(),
 			EnvName:      defaultEnvironmentName,
-			EnvProfile:   "default",
+			EnvProfile:   vars.profile,
 			IsProduction: false,
 		},
 		envCreator:    ssm,
 		projectGetter: ssm,
-		envDeployer:   deployer,
-		projDeployer:  deployer, // TODO #317
+		projDeployer:  deployer,
 		profileConfig: cfg,
 		prog:          spin,
 		identity:      id,
+
+		initProfileClients: func(o *initEnvOpts) error {
```

Comment:
Not quite sure but maybe we can define this function outside of the struct to make it a variable so that we can reuse that?

Reply:
:+1: yeah like @iamhopaul123 says, we can do:

```go
var initEnvProfileClients = func(o *initEnvOpts) error {
   // your function
}
```
and then while initializing the structs:
```go
   initProfileClients: initEnvProfileClients,
```

Reply:
Done in the latest commit.